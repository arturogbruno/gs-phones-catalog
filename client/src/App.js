import React, { useEffect } from "react";
import { Switch, Route } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { addProducts } from "./redux/actions";
import PhonesServices from "./services/PhonesServices";
import Catalog from "./components/Catalog";
import ProductDetail from "./components/ProductDetail";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";
import "./App.css";

function App() {
    const category = "phones";

    const phonesServices = new PhonesServices();

    const dispatch = useDispatch();
    const products = useSelector(state => state.products);

    useEffect(() => {
        const getProductsData = async () => {
            const phones = await phonesServices.getAllPhones();
            dispatch(addProducts(phones));
        }
        getProductsData();
    }, [])

    console.log(products);

    return (
        <div className="App">
            <Navbar category={category} />
            <Switch>
                <Route exact path="/" render={() => <Catalog category={category} products={products} />} />
                <Route exact path={`/${category}/:id`} component={ProductDetail} /> 
            </Switch>
            <Footer />
        </div>
    )
}

export default App;