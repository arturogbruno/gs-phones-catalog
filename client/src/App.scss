@import "./styles//reset.css";

.App {
    background-color: #fff;
    background-image: url("./background.svg");
    background-attachment: fixed;
    background-size: cover;
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 100vh;
    position: relative;
}