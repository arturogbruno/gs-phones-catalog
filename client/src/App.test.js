import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import App from "./App";

describe("testing App", () => {
    let wrapper;
    let store;
    
    beforeEach(() => {
        const mockStore = configureMockStore();

        store = mockStore({
            products: [
                { 
                    id: 2,
                    name: "iPhone 7"
                }
            ]
        });

        wrapper = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );
    });

    it("should render App", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render a Navbar component", () => {
        expect(wrapper.find("Navbar").length).toEqual(1);
    });
});