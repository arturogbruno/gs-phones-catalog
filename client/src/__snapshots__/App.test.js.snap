// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App should render App 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App>
        <div
          className="App"
        >
          <Navbar
            category="phones"
          >
            <div
              className="Navbar"
            >
              <Link
                to="/"
              >
                <LinkAnchor
                  href="/"
                  navigate={[Function]}
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <img
                      alt="GS logo"
                      className="logo"
                      src="https://images.ctfassets.net/5gv1edeicqfs/48EM0LU3Z6gWkQCcCaeoq2/704ea273b5d50d09ff450a5ceaa74631/guidesmiths-logo.png"
                    />
                    <h2>
                      Phones
                       Catalog
                    </h2>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </Navbar>
          <Switch>
            <Route
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
              render={[Function]}
            >
              <Catalog
                category="phones"
                products={
                  Array [
                    Object {
                      "id": 2,
                      "name": "iPhone 7",
                    },
                  ]
                }
              >
                <div
                  className="Catalog"
                >
                  <header>
                    <h1>
                      Welcome to the smallest phones catalog
                    </h1>
                    <h3>
                      We have narrowed the search for you to make your choice easier!
                    </h3>
                  </header>
                  <ProductOverview
                    category="phones"
                    key="2"
                    product={
                      Object {
                        "id": 2,
                        "name": "iPhone 7",
                      }
                    }
                  >
                    <div
                      className="ProductOverview"
                    >
                      <Link
                        to="/phones/2"
                      >
                        <LinkAnchor
                          href="/phones/2"
                          navigate={[Function]}
                        >
                          <a
                            href="/phones/2"
                            onClick={[Function]}
                          >
                            <img
                              alt="undefined iPhone 7"
                              src="images/undefined"
                            />
                            <div
                              className="productMainInfo"
                            >
                              <div
                                className="makerModel"
                              >
                                <h2 />
                                <h2>
                                  iPhone 7
                                </h2>
                              </div>
                              <h2
                                className="price"
                              >
                                 €
                              </h2>
                            </div>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </div>
                  </ProductOverview>
                </div>
              </Catalog>
            </Route>
          </Switch>
          <Footer>
            <div
              className="Footer"
            >
              <p>
                © 2020 Catalog - React Test | Arturo G. Bruno
              </p>
            </div>
          </Footer>
        </div>
      </App>
    </Router>
  </BrowserRouter>
</Provider>
`;
